extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,609 @@
+	}
+
+	/**
+	 * @param $html
+	 * @return string
+	 */
+	private function normalizeHTML( $html ) {
+		if ( empty( $hrml ) ) {
+			return '';
+		}
+
+		$DOM = new DOMDocument( '1.0' );
+		$DOM->formatOutput = true;
+		$DOM->preserveWhiteSpace = false;
+		$DOM->loadXML( $html );
+
+		return $DOM->saveXML();
+	}
+
+	/**
+	 * @param $input
+	 * @param $expectedOutput
+	 * @param $description
+	 * @param $mockParams
+	 * @dataProvider testRenderInfoboxDataProvider
+	 */
+	public function testRenderInfobox( $input, $expectedOutput, $description, $mockParams ) {
+		$wrapper = new \Wikia\Util\GlobalStateWrapper( [ 'wgArticleAsJson' => $mockParams['isMercury'] ?? true ] );
+		$this->mockInfoboxRenderServiceHelper( $mockParams );
+
+		$infoboxRenderService = new PortableInfoboxMobileRenderService();
+		$actualOutput = $wrapper->wrap( function () use ( $infoboxRenderService, $input ) {
+			return $infoboxRenderService->renderInfobox( $input, '', '', '', '' );
+		} );
+		$expectedHtml = $this->normalizeHTML( $expectedOutput );
+		$actualHtml = $this->normalizeHTML( $actualOutput );
+
+		$this->assertEquals( $expectedHtml, $actualHtml, $description );
+	}
+
+	public function testRenderInfoboxDataProvider() {
+		return [
+			[
+				'input' => [ ],
+				'output' => '',
+				'description' => 'Empty data should yield no infobox markup',
+				'mockParams' => [ ]
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+							</aside>',
+				'description' => 'Only title',
+				'mockParams' => [ ]
+			],
+			[
+				'input' => [
+					[
+						'type' => 'navigation',
+						'data' => [
+							'value' => 'navigation value',
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<nav class="pi-navigation pi-item-spacing pi-secondary-background pi-secondary-font">navigation value</nav>
+							</aside>',
+				'description' => 'navigation only',
+				'mockParams' => [ ]
+			],
+			[
+				'input' => [
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+								</div>
+							</aside>',
+				'description' => 'Only pair',
+				'mockParams' => [ ]
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					],
+					[
+						'type' => 'image',
+						'data' => [
+							[
+								'alt' => 'image alt',
+								'url' => 'http://image.jpg',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "http://image.jpg" 13 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)